src/js/media-store.js

```diff
@@ -0,0 +1,1567 @@
+      // (before these fed into things like props, methods, and attributeChangedCallback)
+      // we can also e.g. get updates for the stateOwners themselves
+      else if (type === 'mediaelementchangerequest') {
+        updateStateOwners({ media: detail });
+      } else if (type === 'fullscreenelementchangerequest') {
+        updateStateOwners({ fullscreenElement: detail });
+      } else if (type === 'rootnodechangerequest') {
+        updateStateOwners({ rootNode: detail });
+      }
+      // and we can update our default/options values
+      else if (type === 'optionschangerequest') {
+        // Doing a simple impl for now
+        Object.entries(detail ?? {}).forEach(([optionName, optionValue]) => {
+          // NOTE: updating options will *NOT* prompt any state updates.
+          // However, since we directly mutate options, this allows state owners to be
+          // "live" and automatically updated for any other event or similar monitoring.
+          // For a concrete example, see, e.g., the `mediaSubtitlesShowing.stateOwnersUpdateHandlers`
+          // responsible for managing/monitoring `defaultSubtitles` in the `defaultStateFacade`. (CJP)
+          stateOwners.options[optionName] = optionValue;
+        });
+      }
+    },
+    getState() {
+      // return the current state, whatever it is
+      return state;
+    },
+    subscribe(callback) {
+      if (!callbacks.length && monitorStateOwnersOnlyWithSubscriptions) {
+        // When configured to monitor if and only if there's 1+ subscriptions, we need to force
+        // a "change" in our state owners to cause event handlers to get added. This code
+        // ensures that. (CJP)
+        const stateOwnersToKeep = { ...stateOwners };
+        stateOwners.media = undefined;
+        stateOwners.fullscreenElement = undefined;
+        stateOwners.rootNode = undefined;
+        updateStateOwners(stateOwnersToKeep);
+      }
+      callbacks.push(callback);
+      // give the callback the current state immediately so it can get whatever the state is currently.
+      callback(state);
+      return () => {
+        callbacks.splice(callbacks.indexOf(callback), 1);
+        if (!callbacks.length && monitorStateOwnersOnlyWithSubscriptions) {
+          // When configured to monitor if and only if there's 1+ subscriptions, we need to force
+          // a "change" in our state owners to cause event handlers to get removed. This code
+          // ensures that. (CJP)
+          const stateOwnersToKeep = { ...stateOwners };
+          updateStateOwners({
+            media: undefined,
+            fullscreenElement: undefined,
+            rootNode: undefined,
+          });
+          stateOwners.media = stateOwnersToKeep.media;
+          stateOwners.fullscreenElement = stateOwnersToKeep.fullscreenElement;
+          stateOwners.rootNode = stateOwnersToKeep.rootNode;
+        }
+      };
+    },
+  };
+};
+
+export default createMediaStore;
```

Comment:
I'm seeing how this achieves the general goal. It's also a daunting file and hard to wrap my head fully around despite (or maybe because of) having written half the code in it. Let's find some time to chat through it and see if we can simplify the concepts or organization some.